### ISSUE TITLE
Split JavaScript source code into multiple files

### ISSUE THREAD

**Issue by jadjoubran** [kasperweidick/calculator#13]
Splitting would be a nice learning experience
Normally you'd need webpack to support it, but most browsers now support script type module ([caniuse](https://caniuse.com/#search=module)) so just change your `<script src="">` into a `<script type="module" src="">` and then you'd be able to use import/export

so keep the main JavaScript file that then imports soure code from other files, for example:

```javascript
import {sum} from "./calculations.js"
```

**Comment by jadjoubran** [comment]
I'd recommend doing this task after finishing #12

**Comment by kasperweidick** [comment]
So i tried splitting my main Javascript file into several and then only use the main to import them all. Like: 
`import './defining_variables'`
But that doesn't seem to work (i also changed the script tag as you describe). Does this import/export only work with functions or can you do it (also does it make sense) with a whole file like i tried? Like one file where I define all the variables that will be used

**Comment by jadjoubran** [comment]
yeah if you're reusing it again in a different file, you could define it there
It is quite tough to separate it in multiple files with this structure, I think it'll be easier if the code was organized in classes (but no need to do it if you want to move to your next challenge) if you want

After  we merge this, please ping me again so ican take a look at the overall structure, I'll tell u if something makes sense in a different file

regarding the default export, I think it's because the only thing you're exporting.. but you could also disable that and keep it as a named export
it's fine both ways for this project 👍

**Comment by kasperweidick** [comment]
Ok, it has been merged :-)

**Comment by kasperweidick** [comment]
Even though I also want to move on to other projects now it still think it would be good to discuss the structure a bit. So I learn how it should actually be done.

**Comment by jadjoubran** [comment]
Cool!

Yes so for most of the event handlers (especially the long ones) you could move them into a separate file, such as this one:

`equals.addEventListener("click", () => {`
so it becomes
`equals.addEventListener("click, handleEquals)` where handleEquals is imported

and same for `element.addEventListener("click", (event) => {` (that's for one of the operators)

**Comment by kasperweidick** [comment]
Returned to this after quite a while (mostly to make sure that it still works). But why does my exports and imports not work now?

<img width="1439" alt="calculator" src="https://user-images.githubusercontent.com/43140588/50830901-7ea82580-1349-11e9-8a72-7851c081d4d7.png">

**Comment by kasperweidick** [comment]
Nevermind, i figured it out :-)
